x-pack/agent/docs/agent_configuration_example.yml

```diff
@@ -0,0 +1,518 @@
+           dataset: system.filesystem
+         - metricset: raid
+           dataset: system.raid
+         - metricset: socket
+           dataset: system.socket
+         - metricset: service
+           dataset: system.service
+         - metricset: fsstat
+           dataset: system.fsstat
+         - metricset: foo
+           dataset: system.foo
+
+
+  #################################################################################################
+  # Elasticsearch package example
+  - id?: my-endpoint
+    title: Collect Elasticsearch information
+    package:
+      name: epm/elsticsearch
+      version: 1.7.0
+    inputs:
+        - type: log
+          streams:
+          - id?: {id}
+            enabled?: true # default to true
+            dataset: elasticsearch.audit
+            paths: [/var/log/elasticsearch/*_access.log, /var/log/elasticsearch/*_audit.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.deprecation
+            paths: [/var/log/elasticsearch/*_deprecation.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.gc
+            paths: [/var/log/elasticsearch/*_gc.log, /var/log/elasticsearch/*_gc.log.[0-9]*]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.server
+            paths: [/var/log/elasticsearch/*.log]
+          - id?: {id}
+            enabled?: true
+            dataset: elasticsearch.slowlog
+            paths: [/var/log/elasticsearch/*_index_search_slowlog.log, /var/log/elasticsearch/*_index_indexing_slowlog.log]
+        - type: metrics/elasticsearch
+          hosts: ["http://localhost:9200"]
+          # api_key: "xxxx"
+          xpack.enabled: true
+          streams:
+          - id?: {id}
+            metricset: ccr
+            dataset: elasticseach.ccr
+          - id?: {id}
+            metricset: cluster_stats
+            dataset: elasticseach.cluster_stats
+          - id?: {id}
+            metricset: enrich
+            dataset: elasticseach.enrich
+          - id?: {id}
+            metricset: index
+            dataset: elasticseach.index
+          - id?: {id}
+            metricset: index_recovery
+            dataset: elasticseach.index_recovery
+          - id?: {id}
+            metricset: index_recovery
+            dataset: elasticseach.index_recovery
+          - id?: {id}
+            metricset: ml_jobs
+            dataset: elasticseach.ml_jobs
+          - id?: {id}
+            metricset: node_stats
+            dataset: elasticseach.node_stats
+          - id?: {id}
+            metricset: shard
+            dataset: elasticseach.shard
+
+  #################################################################################################
+  # AWS module
+  - id?: my-aws
+    title: Collect AWS
+    package:
+      name: epm/aws
+      version: 1.7.0
+    inputs:
+        # Looking at the AWS modules, I believe each fileset need to be in their own
```

Comment:
If possible, I think we should make this 1 input with multiple streams. Why do you think it should be multiple inputs?

Reply:
@ruflin From our previous discussion, inputs would contain hosts/ssl configuration. In that context looking at the [filebeat's AWS module](https://github.com/elastic/beats/blob/master/x-pack/filebeat/modules.d/aws.yml.disabled), the events are read from files inside an s3 bucket. 

To fix that problem either:

1. We do what I've done, where multiple inputs are defined where each input have access to a different S3 bucket.
2. Or we allow the S3 input to ignore keys or consider keys that match a certain pattern. (This would require a bit of change in the s3 input, looking at the [configuration](https://github.com/elastic/beats/blob/8b93df1d714a467679f35dda1b476717103f8969/x-pack/filebeat/input/s3/config.go#L15-L22) its not supported)

We do have to think about the use-case, what it's more common? Use a single bucket for all the logs from AWS service or different buckets. I think because of policy/permission the different bucket route might be more popular. 


I am no expert in AWS @kaiyan-sheng might know more here about the usage.

Reply:
Is the queue url defining from where the events are fetched in the end? If yes, shouldn't this be on the stream level instead?

Reply:
For AWS logs, I think better setup is to have different S3 buckets for different logs or different services. For example, one bucket for s3access log, one bucket for elb logs. Or one for vpc flow logs from VPC1 and one for vpc flow logs from VPC2. But all these buckets can have the same notification setup with the same SQS queue(if they are in the same region). 

The queue url is not where the logs are fetched, it only points to the S3 bucket where the actual log needs to be read.

Reply:
How does the input then now, which logs to pick and where to send it to?

Reply:
@ruflin The input then uses the message from SQS queue to locate where the log is(in which s3 bucket and what's the name of the log) and read from S3.

Reply:
I got that part. But it means, each dataset needs to have a different SQS queue as otherwise one dataset would pick up the logs from the others too.

Reply:
@kaiyan-sheng Yes, it know where (bucket) and which (the file) but there is nothing I can see in the code that says: **OK this file is actually a VPCflow**.


So the way I am understand it is the AWS modules assumes that all the events coming from a specific SQS queue can only be the same type. I can be wrong but I don't see how disambiguation could happen.

Reply:
Ahh sorry you are right! With different kinds of logs, they should be separated to different queues if we want to ingest them at the same time.